common/src/KokkosFFT_utils.hpp

```diff
@@ -135,15 +135,19 @@ std::size_t get_index(ContainerType& values, const ValueType& value) {
 }
 
 template <typename T, std::size_t... I>
-std::array<T, sizeof...(I)> make_sequence_array(std::index_sequence<I...>) {
+constexpr std::array<T, sizeof...(I)> make_sequence_array(
+    std::index_sequence<I...>) {
   return std::array<T, sizeof...(I)>{{I...}};
 }
 
-template <int N, typename T>
-std::array<T, N> index_sequence(T const& start) {
-  auto sequence = make_sequence_array<T>(std::make_index_sequence<N>());
-  std::transform(sequence.begin(), sequence.end(), sequence.begin(),
-                 [=](const T sequence) -> T { return start + sequence; });
+template <typename IntType, std::size_t N, int start>
+constexpr std::array<IntType, N> index_sequence() {
+  static_assert(std::is_integral_v<IntType> && std::is_signed_v<IntType>,
+                "index_sequence: IntType must be a signed integer type.");
+  std::array<IntType, N> sequence{};
+  for (std::size_t i = 0; i < N; ++i) {
+    sequence[i] = static_cast<IntType>(start) + static_cast<IntType>(i);
+  }
   return sequence;
 }
```

Comment:
```suggestion
template <typename IntType, std::size_t N, IntType start>
constexpr std::array<IntType, N> index_sequence() {
  static_assert(std::is_integral_v<IntType> && std::is_signed_v<IntType>,
                "index_sequence: IntType must be a signed integer type.");
  std::array<IntType, N> sequence{};
  for (std::size_t i = 0; i < N; ++i) {
    sequence[i] = start + static_cast<IntType>(i);
  }
  return sequence;
}
```

I don't think we need to make a distinction between the two types ?

Reply:
True. Fixed accordingly